fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1284,20 +1284,49 @@ private QueryComponent normalizeAndOr(AndComponent and) {
             QueryComponent child1 = and.getChildren().get(0);
             QueryComponent child2 = and.getChildren().get(1);
             if (child1 instanceof OrComponent && Query.isSingleFieldComparison(child2)) {
-                return OrComponent.from(distributeAnd(child2, ((OrComponent)child1).getChildren()));
+                return OrComponent.from(distributeAnd(Collections.singletonList(child2), ((OrComponent)child1).getChildren()));
             }
             if (child2 instanceof OrComponent && Query.isSingleFieldComparison(child1)) {
-                return OrComponent.from(distributeAnd(child1, ((OrComponent)child2).getChildren()));
+                return OrComponent.from(distributeAnd(Collections.singletonList(child1), ((OrComponent)child2).getChildren()));
             }
         }
         return and;
     }
 
-    private List<QueryComponent> distributeAnd(QueryComponent component, List<QueryComponent> children) {
+    private QueryComponent normalizeAndOrForInAsOr(@Nonnull QueryComponent component) {
```

Comment:
For master (versus patch release), do we want this to be the only definition of `normalizeAndOr`? Or just live with this until a more major overhaul?

We could also take this as a clean merge from the release branch and do that separately.

Reply:
I'd be up for changing this in master, but I think I'd rather do a clean merge. I opened #894 to track this.